components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java

```diff
@@ -51,18 +53,27 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
             // If multi attribute login is enabled, resolve the user before sending recovery notification sending.
             if (IdentityRecoveryServiceDataHolder.getInstance().getMultiAttributeLoginService()
                     .isEnabled(user.getTenantDomain())) {
+                String userDomainQualifiedUsername;
+                if (StringUtils.isNotBlank(user.getRealm())) {
+                    //The PRIMARY domain is not appended by UserCoreUtil.addDomainToName() method. Adding it here.
+                    //This is done to avoid user being resolved when PRIMARY is wrongly passed as the user's realm.
```

Comment:
```suggestion
                    /*
                    The PRIMARY domain is not appended by UserCoreUtil.addDomainToName() method. Adding it here.
                    This is done to avoid user being resolved when PRIMARY is wrongly passed as the user's realm.
                    */
```